### PR TITLE
Nvme Generic Submission Queue Entry

### LINKED ISSUE #46: Import generic submission queue entry
Import & format GSQE structures from SPDK

### PR DESCRIPTION
**Pull request by harloff-g**
Closes #46 

Imports generic command structure from SPDK.
Configures Bazel build environment to include nvme_defs.h

### CODE REVIEW

File: third_party/spdk_defs/nvme_defs.h [hunk 1]
```diff
@@ -243,7 +243,41 @@ struct SglDescriptor {
...
+    } prp;
+    SglDescriptor sgl_descriptor;
+  } dptr;
+  // dword 10-15
+  uint32_t cdw10 : 32;  // command-specific
+  uint32_t cdw11 : 32;  // command-specific
+  uint32_t cdw12 : 32;  // command-specific
+  uint32_t cdw13 : 32;  // command-specific
+  uint32_t cdw14 : 32;  // command-specific
+  uint32_t cdw15 : 32;  // command-specific
```
Comment (basimsahaf) [inline third_party/spdk_defs/nvme_defs.h:276]: nit: we can put these in an array of `uint32_t`s

Comment (harloff-g) [reply third_party/spdk_defs/nvme_defs.h:276]: Good idea

File: third_party/spdk_defs/nvme_defs.h [hunk 2]
```diff
@@ -243,7 +243,36 @@ struct SglDescriptor {
 } ABSL_ATTRIBUTE_PACKED;
 static_assert(sizeof(SglDescriptor) == 16);
 
-} // namespace nvme_defs
+// NVMe Base Specification Figure 7
+// https://nvmexpress.org/wp-content/uploads/NVM-Express-1_4-2019.06.10-Ratified.pdf
```
Comment (eyhhuang) [inline third_party/spdk_defs/nvme_defs.h:247]: i think this should be the Revision 1.0e reference

File: third_party/spdk_defs/nvme_defs.h [hunk 3]
```diff
@@ -243,7 +243,36 @@ struct SglDescriptor {
...
+struct GenericQueueEntryCmd {
+  // dword 0
```
Comment (eyhhuang) [inline third_party/spdk_defs/nvme_defs.h:249]: I looked at the reference and it seems like dwords should be listed in big endian?

Comment (harloff-g) [reply third_party/spdk_defs/nvme_defs.h:249]: Sorry, I messed up my documentation versions causing some confusion. Table 105 at https://nvmexpress.org/wp-content/uploads/NVM-Express-1_4a-2020.03.09-Ratified.pdf should clear this up

**Pull request merged**